src/module.c

```diff
@@ -8458,14 +8460,19 @@ int RM_ACLCheckKeyPermissions(RedisModuleUser *user, RedisModuleString *key, int
     return REDISMODULE_OK;
 }
 
-/* Check if the pubsub channel can be accessed by the user, according to the ACLs associated with it.
- * Glob-style pattern matching is employed, unless the literal flag is
- * set.
+/* Check if the pubsub channel can be accessed by the user based off of the given
+ * access flags. See RM_ChannelAtPosWithFlags for more information about the
+ * possible flags that can be passed in.
```

Comment:
This is inconsistent with the RM_ACLCheckKeyPermissions(), which has a separate ACLREAD/ACLWRITE permission. @oranagra If this makes sense to you, I think we should update the key permission check function to take in RO/RW/UPDATE/etc flags.

Reply:
I'm not certain.
on one hand, it makes sense for RM_ACLCheckKeyPermissions to take ACL (key-spec) flags.
on the other hand, it makes sense to take the same flags used by RM_OpenKey.
@yossigo WDYT?

Reply:
From a module perspective, the flags should describe the operation that is about to be performed - so the current state actually makes more sense to me.

Reply:
@yossigo when you refer to the current state being ok, you mean the current state of the channels API (in this PR), or the keys API (in unstable)?

the issue with RM_OpenKey flags behind provided to RM_ACLCheckKeyPermissions is that the resolution is lower. i.s. it can't specify some of the fine details the key-specs provide (which the ACL currently ignores, but may use someday in the future), like insert / delete, and the difference between READ and ACCESS (excluding metadata).

Reply:
Yeah, I didn't quite follow the recommendation. I assume the current PR is okay for channels though, since there is no corresponding RM_*Channel operations.

Reply:
Sorry about being unclear, I guess I really have to stop trying to multi-task... 😟

I meant the current state of the PR is good and that it makes sense for `RM_ACLCheckChannelPermissions` to refer to channel-specific flags and be inconsistent with keys. As for `RM_ACLCheckKeyPermissions`, I prefer to have a separate set of flags (and extend them as @madolson suggests) and not rely on the `RM_OpenKey` flags.

Reply:
ok. @madolson i suppose we can make that change as part of this PR?